### PR TITLE
feat(Pipes): rename to Pipes and add Pipes.append to extend registry

### PR DESCRIPTION
**Pull request by jeffbcross**

### CODE REVIEW

**Comment by jeffbcross** [comment]
I guess the `refactor(Pipes)` commit message should be changed to `refactor(change_detection)`

File: modules/angular2/test/change_detection/pipes/pipes_spec.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      expect(() => r.get("type", "some object"))
+          .toThrowError(`Cannot find 'type' pipe supporting object 'some object'`);
+    });
+
+    describe('.append()', () => {
+      it('should create a factory that appends new pipes to old', () => {
+        firstPipeFactory.spy("supports").andReturn(false);
+        secondPipeFactory.spy("supports").andReturn(true);
+        secondPipeFactory.spy("create").andReturn(secondPipe);
+        var originalPipes: Pipes = new Pipes({'async': [firstPipeFactory]});
```
Comment (vsavkin) [inline modules/angular2/test/change_detection/pipes/pipes_spec.ts]: I don't think we should type local variables. TS infers all these types automatically.

Comment (jeffbcross) [inline modules/angular2/test/change_detection/pipes/pipes_spec.ts]: True for this line. Not true for line 86.

**Comment by vsavkin** [comment]
@jeffbcross all pipe-related work goes into the pipes namespace, so you should use `fix(pipes), feat(pipes), etc`

**Comment by jeffbcross** [comment]
Merged 8b3efdf229bfc9f1dd0f212845f31dfa82b0863d 9a70f84e600df36670a1bd7e4a7969d3e60a3920 1eebceab2734c31eafa304f0a13d8b3bbcd9f7ea

**Pull request closed without merging**